Change marker-output and sourceFrame-String

fzachert opened the pull request:
Hi,

I made some changes, to make the interface of this detection-task more similar to the aruco-detector-task.
1. The output is now a single RigidbodyState and a vector of all detected RigidBodyStates. In the old version, the output was simply the Pose of the first detected Marker, even there were multiple marker in the image.
2. The output-pose of the marker is now the pose of the marker with respect to the camera. In the old version, the output was inversed and was the camera-pose with respect to the marker (which violates the port-documentation in the orogen-file)
3. I added the marker-ID to the sourceFrame-String of the RigidBodyState. The string is now "apriltag_id_X_frame", where X is the ID of the marker. This is similar to the output of the aruco-Task. In the old version, there was only a constant string with no informations about the marker-ID.

fzachert committed:
fix opencv-version-bug and add marker-id-output

fzachert committed:
Merge branch 'master' of https://github.com/dbcesar/perception-orogen-apriltags

Conflicts:
	tasks/Task.cpp

saarnold committed:
write out marker in camera-frame

saarnold committed:
fixed output of multiple markers

fzachert committed:
Fix rbs-string

dbcesar commented on apriltags.orogen:
```diff
@@ -46,10 +46,11 @@ task_context "Task" do
         .doc 'It contains the calculated marker poses in respect to camera frame of all visible markers'
+#    output_port("marker_poses", "base::samples::RigidBodyState")
+#        .doc 'It contains the calculated marker poses in respect to camera frame of all visible markers'
+    output_port("marker2cam", "base::samples::RigidBodyState")
```
Please add a short description of this output port and you can delete the commented lines.

dbcesar commented on tasks/Task.cpp:
```diff
@@ -234,7 +234,17 @@ void Task::updateHook()
+                    base::samples::RigidBodyState marker2cam = rbs_vector[i];
+                    marker2cam.setTransform(  rbs_vector[i].getTransform().inverse() );
+                    marker_poses.push_back(marker2cam);
+                }
```
Please check the indentation here.

dbcesar commented:
Hi @fzachert , thank you for the improvements, all of them are quite relevant. Please check the comments above before I merge it.  Regards,

fzachert committed:
Some cleanup:
- Add documentation and change name of marker2cam-port to single_marker_pose
- Fix indentation
- Remove unused property source_frame

fzachert commented:
Okay, I fixed the indentation and did some documentation for the new output-port (which has now a more descriptive name).
I also removed the sourceFrame-property, because it is now unused (The source-Frame-string is now created by the marker-ID)

dbcesar commented:
Merged! Thanks!
